modules/internal/String.chpl

```diff
@@ -1532,6 +1532,30 @@ module String {
       return _join(S);
     }
 
+    /*
+      Same as the varargs version, but with all the strings in a list.
+
+      :arg listToJoin: list containing elements whose string values will be concatenated
+
+      :returns: A new string, which is the concatenation of all of the strings
+      passed in with the receiving string inserted between them.
+
+      .. code-block:: chapel
+
+          var someList = new list(string);
+          someList.append("a");
+          someList.append("10");
+          someList.append("d");
+          var x = "|".join(someList);
+          writeln(x); //prints "a|10|d"
+    
+    */
+
+    pragma "no doc"
+    proc join(listToJoin: list(string)) : string {
```

Comment:
This is great, exactly what I had in mind! However...

We need to add at least one test of the new behavior before I can merge this branch. Fortunately, you already have some good testing code in your comment above. That should serve as a good starting point.